ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/ResponseObserverTest.kt

```diff
@@ -92,4 +96,43 @@ class ResponseObserverTest : ClientLoader() {
             assertFalse { observerCalled }
         }
     }
+
+    @Test
+    fun testSavedResponseCanBeReadMultipleTimes() = testWithEngine(MockEngine) {
+        val bodyContent = "Hello"
+        var wasSaved: Boolean? = null
+
+        suspend fun assertBodyCanBeReadMultipleTimes(response: HttpResponse) {
+            assertEquals(bodyContent, response.bodyAsText(), "First read failed")
+            assertEquals(bodyContent, response.bodyAsText(), "It should be possible to read body multiple times")
+        }
+
+        config {
+            install(ResponseObserver) {
+                onResponse { response ->
+                    wasSaved = response.isSaved
+                    // In the response observer itself
+                    assertBodyCanBeReadMultipleTimes(response)
+                }
+            }
+
+            engine {
+                addHandler { respondOk(bodyContent) }
+            }
+        }
+
+        test { client ->
+            // In the pipeline after ResponseObserver
+            client.receivePipeline.intercept(HttpReceivePipeline.After) { response ->
+                assertBodyCanBeReadMultipleTimes(response)
+                proceedWith(response)
+            }
+
+            val response = client.get("/")
+            assertTrue(wasSaved == true)
```

Comment:
looks weird.
Let's split this assertNotNull and assertTrue, so when it fails we see where